pyproject.toml

```
@@ -19,7 +19,7 @@ python = ">=3.9"
 certifi = "2025.1.31"  # See https://github.com/certifi/python-certifi/issues/349
 cryptography = ">=40"
 snowflake-sqlalchemy = "==1.7.3"
-snowflake-connector-python = { version = "<4.0.0", extras = ["secure-local-storage"] }
+snowflake-connector-python = { version = "~=3.15.0", extras = ["secure-local-storage"] }
```

Won't this now only accept patch version changes i.e. `3.15.*`?

```suggestion
snowflake-connector-python = { version = "~=3.15", extras = ["secure-local-storage"] }
```

---

I guess it's a balance between stability and flexibility.